createEnvironment.js

```diff
@@ -295,7 +295,9 @@ function createEnvironment({ baseEnvironment } = {}) {
 
         if (name.includes("failure")) {
           if (event.error) {
-            this.Sentry.captureException(event.error);
+            this.Sentry.captureException(event.error, {
+              tags: this.options.transactionOptions?.tags,
```

Comment:
Hmmm does this actually work? We lack API docs and looking at the types we get:

https://github.com/getsentry/sentry-javascript/blob/01de9df3e45dab3ad6d3c193fce13afa6dd712f2/packages/types/src/client.ts#L31

which `EventHint` is: https://github.com/getsentry/sentry-javascript/blob/01de9df3e45dab3ad6d3c193fce13afa6dd712f2/packages/types/src/event.ts#L76-L84

Also `transactionOptions` is a bit poorly named now that we are tagging error events as well, but that's probably fine for now

Reply:
It worked when I tested it locally, although the typing in vscode wasn't `EventHint`, it was `CaptureContext`, so I'm not sure why it works: https://github.com/getsentry/sentry-javascript/blob/develop/packages/types/src/scope.ts#L13-L26

Another option is to define the tags globally with `this.Sentry.setTags(this.options.transactionOptions?.tags)`: https://github.com/getsentry/sentry-javascript/blob/01de9df3e45dab3ad6d3c193fce13afa6dd712f2/packages/core/src/exports.ts#L119

This works that same when testing locally and is arguably a little better as it should ensure future changes would get the tags as well.

Reply:
Yeah I think using `setTags` would be better.

What do you think about trying to load tags from `this.options.tags` and falling back to `transactionOptions.tags`? e.g. `this.Sentry.setTags(this.options.tags || this.options.transactionOptions?.tags)`

At some other point we can make the breaking change to remove `transactionOptions.tags`.

Reply:
I just tried out using setTags and that will add tags to both the exception and the transcations, so I've reduced this PR to just the setTags() call as this would make the breaking change easier in the future (only one place to cleanup in this codebase).

Let me know if you'd prefer to keep the tags in the startTransaction calls and if so.

Reply:
Nope, this is perfect, thanks.